### ISSUE TITLE
XCode 15.0 Release - firebase-ios-sdk bug: An attribute list cannot appear here

### ISSUE THREAD

**Issue by dsert1** [firebase/firebase-ios-sdk#11840]
### Description

The FIRFirestoreSettings has an error after updating to XCode release version 15.0 on this line: 
`// Public constant
ABSL_CONST_INIT extern "C" const int64_t kFIRFirestoreCacheSizeUnlimited =
    Settings::CacheSizeUnlimited;`
Error: `An attribute list cannot appear here`

    
    

### Reproducing the issue

Try to build an existing iOS app that was previously working that used the `FirebaseFirestore` module. It no longer builds after the XCode update. 

### Firebase SDK Version

??

### Xcode Version

15.0

### Installation Method

Swift Package Manager

### Firebase Product(s)

Firestore

### Targeted Platforms

iOS

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by dconeybe** [comment]
@dsert1 Thank you for reporting this issue. However, I am not seeing the same issue when I compile using Xcode 15. If possible, could you do the following:

1. Try to compile the "quickstart" Firestore application using your Xcode 15 setup. To do this, clone the https://github.com/firebase/quickstart-ios repository and attempt to compile and build the application in the "firestore" subidrectory.
2. Provide a screenshot or full log of the compilation error.
3. Try to figure out which version of the Firebase iOS SDK that you are using (e.g. the latest at the time of writing is 10.15.0).

Thank you.

**Comment by rharshaw** [comment]
@dsert1 I had the same issue. I think you may be operating on Firebase 9.6.0 SDK version.

I fixed the issue by going into my Project - Package Dependencies.

I changed the range from 9.0.0 < 10.0.0 to 10.0.0 < 11.0.0

Then changed the drop down again to 'Update to Next Major Version'. Once that set in, the packages updated and my project ran fine. 

Hope that helps.

**Comment by kemalekren** [comment]
I got the same issue as well.  Here is the details of my side:
I'm using SPM - Firebase 9.6.0
**Error Line:** 
`ABSL_CONST_INIT extern "C" const int64_t kFIRFirestoreCacheSizeUnlimited =
    Settings::CacheSizeUnlimited;`
**Error:** `An attribute list cannot appear here`
**Error file path** :  Firebase -> Firestore -> API -> FIRFirestoreSettings  / Line - 33
**Running Device:** Simulator 14 Pro / 16.4

**Comment by dconeybe** [comment]
For anyone experiencing this issue, please attempt to reproduce in the latest version, which is 10.15.0 at the time of writing. If this is fixed in the latest version then the "fix" is to upgrade. Otherwise, we will fix it in the next version. So either way, an upgrade will be needed in order to get the fix. Thanks!

**Comment by Resource-2** [comment]
any update for this bug.

**Comment by dconeybe** [comment]
@Resource-2 What version of the Firebase SDK are you using? It looks like this problem only occurs with old versions of the SDK. Does the problem go away when you upgrade to the latest version, which is 10.15.0 (at the time of writing)?

**Comment by Marvelousy20** [comment]
@rharshaw solution worked for me.

**Comment by dconeybe** [comment]
I am going to close this issue since there have been several reports that upgrading to the latest version of the Firebase SDK has fixed the problem. If anyone experiences this issue with the latest version, which is 10.15.0 at the time of writing, or newer, then please leave a comment or open a new issue and we will investigate.

**Comment by Granluke** [comment]
@kemalekren Did updating help you? I updated to 10.15.0 and still get the same error with 
ABSL_CONST_INIT extern "C" const int64_t kFIRFirestoreCacheSizeUnlimited = Settings::CacheSizeUnlimited;
Error: An attribute list cannot appear here

**Comment by dconeybe** [comment]
I attempted to reproduce by building the Firestore quickstart-ios application, and everything went just fine. That project uses cocoapods (rather than SPM) so maybe this is only a problem when using SPM. Can those of you who are experiencing this bug leave a comment to indicate if you are using CocoaPods or SPM?

For completeness, here is the script I ran in an attempt to reproduce:

```bash
#!/bin/bash

set -xveuo pipefail

which pod
pod --version
sudo xcode-select -s /Applications/Xcode_15.app/Contents/Developer
git clone https://github.com/firebase/quickstart-ios.git
cd quickstart-ios/firestore
cp ~/Downloads/GoogleService-Info.plist .
rm Podfile.lock
pod update
xcodebuild -workspace FirestoreExample.xcworkspace -scheme FirestoreExample -sdk iphonesimulator -destination 'platform=iOS Simulator,name=iPhone 15' CODE_SIGNING_REQUIRED=NO clean build
echo 'SUCCESS!!!!!!!!!!!!!'
```

**Comment by 1512529** [comment]
Any solution without update firebase sdk?

**Comment by huuthanit** [comment]
Any solution without update firebase sdk?

**Comment by huuthanit** [comment]
@dconeybe I using react-native with CocoaPods

**Comment by aslamanver** [comment]
Any solution having issue 

`System:
    OS: macOS 14.0
    CPU: (8) arm64 Apple M1 Pro
    Memory: 82.33 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.16.1 - /opt/homebrew/opt/node@18/bin/node
    Yarn: Not Found
    npm: 9.5.1 - /opt/homebrew/opt/node@18/bin/npm
    Watchman: 2023.06.26.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: 1.13.0 - /Users/aslam/.rvm/gems/ruby-2.7.6/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 23.0, iOS 17.0, macOS 14.0, tvOS 17.0, watchOS 10.0
    Android SDK: Not Found
  IDEs:
    Android Studio: 2022.2 AI-222.4459.24.2221.10121639
    Xcode: 15.0/15A240d - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.18 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: 10.1.3 => 10.1.3 
    react: 18.2.0 => 18.2.0 
    react-native: 0.71.4 => 0.71.4 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found`

**Comment by Granluke** [comment]
I am using firebase with SPM however I also have cocoapods activated for two pods. On 28. Sep 2023, at 06:59, Aslam Anver ***@***.***> wrote:﻿
Any solution having issue
System: OS: macOS 14.0 CPU: (8) arm64 Apple M1 Pro Memory: 82.33 MB / 16.00 GB Shell: 5.9 - /bin/zsh Binaries: Node: 18.16.1 - ***@***.***/bin/node Yarn: Not Found npm: 9.5.1 - ***@***.***/bin/npm Watchman: 2023.06.26.00 - /opt/homebrew/bin/watchman Managers: CocoaPods: 1.13.0 - /Users/aslam/.rvm/gems/ruby-2.7.6/bin/pod SDKs: iOS SDK: Platforms: DriverKit 23.0, iOS 17.0, macOS 14.0, tvOS 17.0, watchOS 10.0 Android SDK: Not Found IDEs: Android Studio: 2022.2 AI-222.4459.24.2221.10121639 Xcode: 15.0/15A240d - /usr/bin/xcodebuild Languages: Java: 11.0.18 - /usr/bin/javac npmPackages: @react-native-community/cli: 10.1.3 => 10.1.3  react: 18.2.0 => 18.2.0  react-native: 0.71.4 => 0.71.4  react-native-macos: Not Found npmGlobalPackages: *react-native*: Not Found

**Comment by cuddimatic** [comment]
in Xcode, go to FIRFirestoneSettings.mm file, look for;-

ABSL_CONST_INIT extern "C" const int64_t kFIRFirestoreCacheSizeUnlimited =
    Settings::CacheSizeUnlimited;

and change it to;-

extern "C" const int64_t kFIRFirestoreCacheSizeUnlimited =
Settings::CacheSizeUnlimited;

run your build, this should work

**Comment by dconeybe** [comment]
The problematic `ABSL_CONST_INIT` was removed in Sept 2022 in https://github.com/firebase/firebase-ios-sdk/pull/10221 and that PR was first included in the Firebase SDK version 10.0.0. Upgrading to a version >= 10.0.0 will fix this issue.

Alternately, as suggested in a previous comment, you can edit the source code in the `Pods` directory and simply remove the `ABSL_CONST_INIT` modifier on `kFIRFirestoreCacheSizeUnlimited`.

I'm going to close this issue because, as far as I can tell, it's fixed and is _impossible_ for this build error to be occurring in the latest version of the Firebase SDK. If you find that this is still occurring with the latest version of the Firebase SDK then please confirm 100% that you are indeed using the latest version. If you are, please provide a GitHub repository that reproduces this build error that I can clone and reproduce for myself.

**Comment by ozgurshn** [comment]
I get this error on Xcode 15 with the latest SDK 10.15.0
<img width="1111" alt="Screenshot 2023-09-28 at 18 48 55" src="https://github.com/firebase/firebase-ios-sdk/assets/2404074/ed018324-78fd-4f48-96aa-a9cfb8f6797c">

**Comment by ozgurshn** [comment]
Thanks @dconeybe  It seems to be after removing and adding FirebaseFirestore, FirebaseFirestoreSwift again in the frameworks,libraries in the project target. I also run pod install but I have different Pods not related to Firebase.

**Comment by victordsantoss** [comment]
@cuddimatic  Good!! It worked here.

**Comment by huuthanit** [comment]
> > in Xcode, go to FIRFirestoreSettings.mm file, look for(go to pods, and you will see FIRbaseFirestore folder, open it);-
> > ```
> > ABSL_CONST_INIT extern "C" const int64_t kFIRFirestoreCacheSizeUnlimited =
> >     Settings::CacheSizeUnlimited;
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > and change it to;-
> > ```
> > extern "C" const int64_t kFIRFirestoreCacheSizeUnlimited =
> > Settings::CacheSizeUnlimited;
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > run your build, this should work
> 
> Can't we make this process from Podfile automate.

This is solution for automate.

on Podfile
`  post_install do |installer|
      system('sh fix-build-for-xcode15.sh')
  end`
  on file fix-build-for-xcode15.sh
  `#issue https://github.com/firebase/firebase-ios-sdk/issues/11840
echo "Fixing build for Xcode 15"
sed -i '' 's/ABSL_CONST_INIT extern "C" const int64_t kFIRFirestoreCacheSizeUnlimited =/extern "C" const int64_t kFIRFirestoreCacheSizeUnlimited =/g' Pods/FirebaseFirestore/Firestore/Source/API/FIRFirestoreSettings.mm
`

**Comment by dconeybe** [comment]
@huuthanit Thanks for the Podfile hack! You can probably even simplify your sed script further by just stripping out all occurrences of `ABSL_CONST_INIT`, something like this:

```bash
sed -i '' 's/ABSL_CONST_INIT//g' Pods/FirebaseFirestore/Firestore/Source/API/FIRFirestoreSettings.mm
```

After all, `ABSL_CONST_INIT` doesn't actually affect any runtime behavior, but rather ensures that the variable can be initialized at compile time with a compile-time constant. It does not affect correctness.

**Comment by Granluke** [comment]
For those of you running SPM and Cocoapods:

I had to disable: ENABLE_USER_SCRIPT_SANDBOXING
Because the true error appeared to be: Sandbox: rsync.samba
I replicated this by creating a new repository and installing all the same swift packages and pods but not initalizing/using firebase.

**Comment by nillbiplob** [comment]
> in Xcode, go to FIRFirestoreSettings.mm file, look for(go to pods, and you will see FIRbaseFirestore folder, open it);-
> 
> ```
> ABSL_CONST_INIT extern "C" const int64_t kFIRFirestoreCacheSizeUnlimited =
>     Settings::CacheSizeUnlimited;
> ```
> 
> and change it to;-
> 
> ```
> extern "C" const int64_t kFIRFirestoreCacheSizeUnlimited =
> Settings::CacheSizeUnlimited;
> ```
> 
> run your build, this should work

This suggestion worked for me as well.

**Comment by alexrybachev** [comment]
The same fatal error. Xcode 15, Firebase SPM

**Comment by alexrybachev** [comment]
@ dconeybe the problem is on. Xcode 15, firebase spm...

**Comment by paulb777** [comment]
As @dconeybe pointed out, the Firebase issue is fixed in versions 10.0.0 and higher. 

I'm going to close for future comments. Scroll up to see workarounds on earlier versions.
